doc/api/modules.md

```diff
@@ -481,81 +481,126 @@ added: v0.1.16
+  module.exports.b = 'Add a props b';
+  module.exports = {
+    c: 'Assign module.exports to a new Object, so a and b are invalid.'
+  };
+  exports.d = 'exports is invalid here, its referred to old module.exports';
+  exports = module.exports; // reassigning exports to new module.exports
+  exports.e = 'After reassigned to new module.exports, exports works again.';
+  //
+  exports = {
+    f: 'Assign exports to a local Object, so its not referred to module.exports'
+  };
+  exports.g = 'This just add a prop to the local Object above.';
+}
 
-y.js:
+const exported = _require(synchronize); // Only c and e will be exported.
+console.log(exported);
+/* The exported result will be:
+{
+  c: 'Assign a new Object to module.exports, so a and b are invalid.'
+  e: 'After reassigned to new module.exports, exports works again.'
+}
+*/
+```
 
+Asynchronous behaviors (directly checking in file) :<br/>
+Assume you have a `x.js`:
 ```js
-const x = require('./x');
-console.log(x.a);
+// Let's name the exported value, to make things clearly.
+const the_returned_exports = {
+  a: 'sync export prop a'
+};
+module.exports = the_returned_exports;
+setTimeout( () => {
+  module.exports.b = 'okay, module.exports is referred to the_returned_exports'+
+  ' which returned in previously tickcount.';
+  module.exports = {
+    c: 'does nothing, it assign a new Object, but the module.exports was' +
+    ' returned at previously tickcount.( the return is the_returned_exports ).'
+  }
+},1000);
 ```
-
-#### exports alias
-<!-- YAML
-added: v0.1.16
--->
-
-The `exports` variable that is available within a module starts as a reference
-to `module.exports`. As with any variable, if you assign a new value to it, it
-is no longer bound to the previous value.
-
-To illustrate the behavior, imagine this hypothetical implementation of
-`require()`:
-
+Then in `y.js`:
 ```js
-function require(...) {
-  // ...
-  ((module, exports) => {
-    // Your module code here
-    exports = some_func;        // re-assigns exports, exports is no longer
-                                // a shortcut, and nothing is exported.
-    module.exports = some_func; // makes your module export 0
-  })(module, module.exports);
-  return module;
-}
+const x = require('./x');
+console.log(x); // { a } here
+setTimeout( () => {
+  console.log(x); // { a, b } here
+},1000);
+// c is not exported
 ```
 
-As a guideline, if the relationship between `exports` and `module.exports`
-seems like magic to you, ignore `exports` and only use `module.exports`.
+There is one more thing to notice: `this`. Because `module.exports` was passed
+as `this` argument for `The module wrapper`, you may used `this` as an alias of
+`exports` to export values previously, But since `this` is an undocumented
+value, the behavior of `this` is not guaranteed. Ex: when you use `babel`,
+`transform-es2015-modules-commonjs` will break `this`, so you should not use
+`this` as a shortcut for export.
```

Comment:
I don't agree with the above. this is documented, its documented right here! And if the module does not work when some random thirdparty tool is used to convert it to some other form... that should be part of that tools documentation, not proposed as a general rule for node modules.